src/outputs/restart.hpp

```diff
@@ -0,0 +1,145 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef OUTPUTS_RESTART_HPP_
+#define OUTPUTS_RESTART_HPP_
+//! \file io_wrapper.hpp
+//  \brief defines a set of small wrapper functions for MPI versus Serial Output.
+#include <cinttypes>
+#ifdef HDF5OUTPUT
+#include <hdf5.h>
+#endif
+#include <vector>
+
+#include "outputs/parthenon_hdf5.hpp"
+namespace parthenon {
+class Mesh;
+
+class RestartReader {
+ public:
+  explicit RestartReader(const char *theFile);
+
+  // Gets single block data for variable name and
+  // fills internal data for given pointer
+  // returns 1 on success, negative on failure
+  //! \fn void RestartReader::ReadBlock(const char *name, const int blockID, T *data)
+  //  \brief Reads data for one block from restart file
+  template <typename T>
+  int ReadBlock(const char *name, const int blockID, T *data, size_t vlen) {
+    IndexRange range{blockID, blockID};
+    return ReadBlocks(name, range, data, vlen);
+  }
+
+
+  // Gets data for all blocks on current rank.
+  // Assumes blocks are contiguous
+  // fills internal data for given pointer
+  // returns NBlocks on success, negative on failure
+  template <typename T>
+  int ReadBlocks(const char *name, IndexRange range, T *data, size_t vlen = 1) {
+#ifdef HDF5OUTPUT
+    try {
+      herr_t status;
+
+      // compute block size, probaby could cache this.
+      hid_t theHdfType = getHdfType(data);
+
+      hid_t dataset = H5Dopen2(fh_, name, H5P_DEFAULT);
+      if (dataset < 0) {
+	return -1;
```

Comment:
I know this isn't your fault, but all these magic numbers are driving me batty. Can we define them at the top of the file, maybe?

Reply:
It should actually throw a parthenon::warn with an appropriate warning message.  In the calling code I wanted to just capture whether it is less than 0 or not. However for debugging the different numbers help narrow down where the failure occurred.  Probably makes sense to create an enum (class?) for the different failure types, but then I'd need special code for the callers rather than a simple check less than 0.  Will look at this, but is lower priority than fixing conflicts with develop.

Reply:
Makes sense.

Sounds like a raw enum might be best here actually, since then it automatically casts, and you don't need special code.

Reply:
Actually now that the code is working, I'm just going to return a boolean.

Reply:
Actually @Yurlungur it **IS** my fault.

Reply:
I've changed this to return either number of blocks read, or -1 on error.